### ISSUE TITLE
A question about implementation

### ISSUE THREAD

**Issue by SydCS** [cs-giung/lipsum-ft#1]
Hello. Thank you for this work!

I have a question about the implementation in Lipsum-FT.py:
`logits = jnp.exp(pretrained_params['logit_scale']) * output @ params['cls']`
`source = jax.nn.log_softmax(logits, axis=-1)`

Why jnp.exp `output @ params['cls']` before log_softmax? From my understanding, the exponential operation of logits is already contained in log_softmax.
Looking forward to your kind reply!

**Comment by cs-giung** [comment]
The reason for applying `jnp.exp(pretrained_params['logit_scale'])` before log_softmax is that it follows the original CLIP implementation (https://github.com/openai/CLIP/blob/dcba3cb2e2827b402d2701e7e1c7d9fed8a20ef1/clip/model.py#L367). In CLIP, logit_scale is stored in a logarithmic form, and its exponentiation is used to scale the logits before computing the softmax.

**Comment by SydCS** [comment]
I see! I must have been out of my mind to misread the scope of the parentheses.
Thank you for this reply.
